### PR TITLE
Fixate the data size @open sesame 10/14 10:07

### PR DESCRIPTION
**Pull request by gichan-jang**
An error occurs in the communication because of different data sizes in the 32-bit and the 64-bit systems. Fixate the data size.

Tested using 32bit Tizen TV and 64bit Ubuntu.

Signed-off-by: gichan <gichan2.jang@samsung.com>

### CODE REVIEW

File: src/libnnstreamer-edge/nnstreamer-edge-aitt.c [hunk 1]
```diff
@@ -221,7 +221,7 @@ _nns_edge_invoke_event_cb (nns_edge_handle_s * eh, nns_edge_event_e event,
  */
 static void
 aitt_cb_message_arrived (aitt_msg_h msg_handle, const void *msg,
-    size_t msg_len, void *user_data)
+    uint64_t msg_len, void *user_data)
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-aitt.c:224]: Q: does AITT use 64b data type for message size? (IMO size_t > unit32_t)

Comment (jaeyun-jung) [reply src/libnnstreamer-edge/nnstreamer-edge-aitt.c:224]: or size_t > unsigned int

Comment (gichan-jang) [reply src/libnnstreamer-edge/nnstreamer-edge-aitt.c]: Thanks. I only modified the data which is related with communication.

File: src/libnnstreamer-edge/nnstreamer-edge-queue.h [hunk 2]
```diff
@@ -25,7 +26,7 @@ typedef void *nns_edge_queue_h;
 /**
  * @brief Callback called when data in queue is released.
  */
-typedef void (*nns_edge_queue_data_destroy_cb) (void *data);
+typedef int (*nns_edge_queue_data_destroy_cb) (void *data);
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-queue.h:29]: Q) why did you change ret type?

File: src/libnnstreamer-edge/nnstreamer-edge-data.c [hunk 3]
```diff
@@ -464,7 +464,7 @@ nns_edge_data_serialize (nns_edge_data_h data_h, void **data, size_t *len)
   }
 
   ret = nns_edge_metadata_serialize (ed->metadata, &meta_serialized,
-      &edata_header.meta_len);
+      (uint32_t *) & edata_header.meta_len);
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-data.c:467]: unnecessary change, edata_header.meta_len is uint32_t

File: src/libnnstreamer-edge/nnstreamer-edge-data.c [hunk 4]
```diff
@@ -542,7 +542,7 @@ nns_edge_data_deserialize (nns_edge_data_h data_h, void *data)
     ptr += ed->data[n].data_len;
   }
 
-  ret = nns_edge_metadata_deserialize (ed->metadata, ptr, meta_len);
+  ret = nns_edge_metadata_deserialize (ed->metadata, ptr, (size_t) meta_len);
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-data.c:545]: how about changing size_t to uint32_t in nns-edge repo?

Comment (gichan-jang) [reply src/libnnstreamer-edge/nnstreamer-edge-data.c]: Updated. thanks.

**Comment by gichan-jang** [comment]
This patch breaks nnstreamer build.

File: src/libnnstreamer-edge/nnstreamer-edge-data.c [hunk 5]
```diff
@@ -542,7 +542,7 @@ nns_edge_data_deserialize (nns_edge_data_h data_h, void *data)
     ptr += ed->data[n].data_len;
   }
 
-  ret = nns_edge_metadata_deserialize (ed->metadata, ptr, meta_len);
+  ret = nns_edge_metadata_deserialize (ed->metadata, ptr, (uint32_t) meta_len);
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-data.c:545]: unnecessary change

File: src/libnnstreamer-edge/nnstreamer-edge-internal.c [hunk 6]
```diff
@@ -451,7 +451,8 @@ _nns_edge_transfer_data (nns_edge_conn_s * conn, nns_edge_data_h data_h,
 
   nns_edge_data_get_count (data_h, &cmd.info.num);
   for (i = 0; i < cmd.info.num; i++)
-    nns_edge_data_get (data_h, i, &cmd.mem[i], &cmd.info.mem_size[i]);
+    nns_edge_data_get (data_h, i, &cmd.mem[i],
+        (uint32_t *) & cmd.info.mem_size[i]);
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-internal.c:455]: IMO unnecessary change

File: tests/unittest_nnstreamer-edge.cc [hunk 7]
```diff
@@ -3232,15 +3233,15 @@ TEST(edgeQueue, pushData)
   for (i = 0; i < 5U; i++)
     EXPECT_EQ (((unsigned int *) result)[i], i + 30U);
 
-  EXPECT_TRUE (nns_edge_queue_push (queue_h, data1, nns_edge_free));
+  EXPECT_TRUE (nns_edge_queue_push (queue_h, data1, (nns_edge_queue_data_destroy_cb) nns_edge_free));
```
Comment (jaeyun-jung) [inline tests/unittest_nnstreamer-edge.cc:3236]: unnecessary?

**Comment by jaeyun-jung** [comment]
We should upload nnstreamer and edge repo to change data type after M2 release.

**Comment by myungjoo** [comment]
Suggestion: size_t --> uint64. We assume 64bit as the default system.

**Comment by gichan-jang** [comment]
> Suggestion: size_t --> uint64. We assume 64bit as the default system.

Thanks. The type of size variable is set to 64bit as the default.

**Review by myungjoo — APPROVED** [review]

**Review by jaeyun-jung — APPROVED** [review]
